server/server.js

```diff
@@ -218,6 +219,30 @@ Server.prototype._handleResourceMessage = function(socket, message, messageType)
   }
 };
 
+// Process the resource messages associated with a single client
+Server.prototype._handleResourceMessages = function (socket, client) {
+  var subscriptions = client.subscriptions;
+  var presences = client.presences;
+
+  // Pause events on the inbound socket
+  Pauseable.pause(socket);
+
+  for (var key in subscriptions) {
+    var message = subscriptions[key];
+    var messageType = this._getMessageType(message.to);
```

Comment:
Yeah. Maybe these messages already have the type in options. But I think a restore is already an expensive operation. Let's get this across for now. 

Also, in a future PR, name_sync messages should carry a copy of all existing client side subs and presences so that we can still keep the client portable across account migration or disaster recovery etc (since we may lose the backend data). They should be compared with what we have on the server side and anything extra should be subscribed to as well.